### ISSUE TITLE
[Offline map imagery] Fetch tiles in parallel to improve offline download performance

### ISSUE THREAD

**Issue by gino-m** [google/ground-android#1704]

**Comment by gino-m** [comment]
@sufyanAbbasi @shobhitagarwal1612 Can you please coordinate one this one?

**Comment by shobhitagarwal1612** [comment]
Duplicate of https://github.com/google/ground-android/issues/2355
